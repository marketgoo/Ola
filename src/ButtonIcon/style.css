.ola_buttonIcon {
    --color: var(--color-neutral-700);
    --background: transparent;
    --color-hover: var(--color-neutral-900);
    --background-hover: var(--color-neutral-200);
    --shadow-focus: var(--color-accent-100);
    --background-active: var(--color-neutral-400);

    display: inline-flex;
    min-height: var(--size-3);
    min-width: var(--size-3);
    align-items: center;
    justify-content: center;
    background: none;
    color: var(--color);
    border: none;
    cursor: pointer;
    padding: 0;
    outline: 0;
    vertical-align: middle;

    &::-moz-focus-inner {
        border: 0;
    }

    & > svg {
        display: block;
        flex-shrink: 0;
        max-height: 100%;
        max-width: 100%;
        border-radius: 50%;
        fill: var(--color);
        background-color: var(--background);
        transition: background-color var(--duration);
    }

    &:focus-visible svg {
        fill: var(--color-hover);
        box-shadow: 0 0 0 4px var(--shadow-focus);
    }

    &:hover svg {
        fill: var(--color-hover);
        background: var(--background-hover);
    }

    &:active svg {
        background: var(--background-active);
    }

    & .ola_icon.is-small {
        padding: var(--size-0);
    }
    & .ola_icon.is-medium,
    & .ola_spinner {
        padding: var(--size-1);
    }

    &.is-secondary {
        --color: var(--color-neutral-400);
    }

    &.is-destructive {
        --color: var(--color-negative-500);
        --shadow-focus: var(--color-negative-100);
        --color-hover: var(--color-negative-600);
        --background-active: var(--color-negative-100);
    }

    &.is-chevron {
        background-repeat: no-repeat;
        background-position: right center;
        padding-right: var(--size-4);
        border-radius: var(--radius-xl);

        /* Chevron gray */
        background-image: url('data:image/svg+xml;charset=UTF-8,<svg width="44px" height="44px" viewBox="0 0 44 44" version="1.1" xmlns="http://www.w3.org/2000/svg"><path d="M22,23.5857864 L27.2928932,18.2928932 C27.6834175,17.9023689 28.3165825,17.9023689 28.7071068,18.2928932 C29.0976311,18.6834175 29.0976311,19.3165825 28.7071068,19.7071068 L22.7071068,25.7071068 C22.3165825,26.0976311 21.6834175,26.0976311 21.2928932,25.7071068 L15.2928932,19.7071068 C14.9023689,19.3165825 14.9023689,18.6834175 15.2928932,18.2928932 C15.6834175,17.9023689 16.3165825,17.9023689 16.7071068,18.2928932 L22,23.5857864 Z" fill="%2357657a"></path></svg>');
    }

    &.is-disabled,
    &:disabled {
        color: var(--color-neutral-200);
        --background: var(--color-white);
        --background-hover: var(--color-white);
        --background-active: var(--color-white);
        cursor: not-allowed;
        pointer-events: none;
        box-shadow: none;

        &.is-chevron {
            /* Chevron gray-xlight */
            background-image: url('data:image/svg+xml;charset=UTF-8,<svg width="44px" height="44px" viewBox="0 0 44 44" version="1.1" xmlns="http://www.w3.org/2000/svg"><path d="M22,23.5857864 L27.2928932,18.2928932 C27.6834175,17.9023689 28.3165825,17.9023689 28.7071068,18.2928932 C29.0976311,18.6834175 29.0976311,19.3165825 28.7071068,19.7071068 L22.7071068,25.7071068 C22.3165825,26.0976311 21.6834175,26.0976311 21.2928932,25.7071068 L15.2928932,19.7071068 C14.9023689,19.3165825 14.9023689,18.6834175 15.2928932,18.2928932 C15.6834175,17.9023689 16.3165825,17.9023689 16.7071068,18.2928932 L22,23.5857864 Z" fill="%23f0f2f4"></path></svg>');
        }
    }

    &.is-busy {
        color: var(--color-white);
        background-color: var(--color-neutral-700);
        min-width: 0;
        cursor: wait;
        pointer-events: none;

        &.is-chevron {
            background-image: none;
        }
    }
}

