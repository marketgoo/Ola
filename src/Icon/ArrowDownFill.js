import React from 'react'

const ArrowDownFill = () => <path fill="currentColor" d="M4.557 12.463A.75.75 0 0 1 5.25 12h13.5a.75.75 0 0 1 .53 1.28l-6.75 6.75a.747.747 0 0 1-1.06 0l-6.75-6.75a.75.75 0 0 1-.163-.817Z"/>

export default ArrowDownFill
