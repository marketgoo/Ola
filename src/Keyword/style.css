.ola_keyword {
    --color: var(--color-neutral-700);
    --background: var(--color-neutral-200);

    display: flex;
    align-items: center;
    color: var(--color);
    background-color: var(--background);
    border-radius: var(--radius-small);
    vertical-align: baseline;
    cursor: default;
    max-width: max-content;
    min-height: var(--size-8);
    letter-spacing: 0;
}

.ola_keyword-text {
    font: var(--font-caption);
    font-weight: var(--bold);
    padding: 0.8em var(--column-gap-1);
}

.ola_keyword-button {
    --color: var(--color);
    flex: 0 0 auto;
    align-self: flex-start;

    & .ola_buttonIcon {
        min-width: var(--size-8);
        min-height: var(--size-8);
        border-radius: var(--radius-small);
        --background-hover: var(--color-neutral-400);
    }
}
