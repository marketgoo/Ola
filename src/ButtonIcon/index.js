import React from 'react'
import {default as PT} from 'prop-types'
import { getElementType } from '../utils'
import cx from 'classnames'
import Icon from '../Icon'
import Spinner from '../Spinner'

const ButtonIcon = ({ as, className, busy, disabled, color, variant, invert, icon, ...props }) => {
  const ElementType = getElementType(ButtonIcon, { as: as, ...props })
  delete props.as

  const variantNumber = variant === 'normal' ? '500' : '900'

  const iconElement = typeof icon === 'string' ? <Icon name={icon} /> : icon

  const styles = cx(
    'ola_button_icon',
    `is-${variant}`,
    {'is-busy': busy},
    {'is-disabled': disabled && !busy},
    {'is-invert': invert},
    className
  )
  return (
    <ElementType
      className={styles}  
      disabled={busy || disabled}
      style={{
        '--base-color': `var(--color-${color}-${variantNumber})`,
        '--busy-color': `var(--color-${color}-600)`,
        '--disabled-color': `var(--color-${color}-300)`,
        '--shadow-focus': `var(--shadow-focus-${color})`,
      }}
      {...props}>
      {busy ? <Spinner /> : iconElement}
    </ElementType>
  )
}

ButtonIcon.defaultProps = {
  as: 'button',
  variant: 'normal',
  color: 'neutral',
  busy: false,
  disabled: false,
  invert: false,
  icon: null
}

ButtonIcon.propTypes = {
  /** Name of the icon in this lib (see Ola Icon), or a svg direcly **(preferred a Phosphor Icon)** */
  icon: PT.oneOfType([
    PT.string,
    PT.shape({
      type: PT.oneOf(['svg']),
    })
  ]),
  /** Render ButtonIcon with any html tag */
  as: PT.string,
  /** Icon color */
  color: PT.oneOf(['primary', 'neutral', 'positive', 'negative', 'warning', 'pro', 'accent']),
  /** Icon variant (normal=color-500, dark=color-900) */
  variant: PT.oneOf(['normal', 'dark']),
  /** Extra className */
  className: PT.string,
  /** Button busy state */
  busy: PT.bool,
  /** Button disabled state */
  disabled: PT.bool,
  /** Indicate if icon should have color white */
  invert: PT.bool,
}

export default ButtonIcon
