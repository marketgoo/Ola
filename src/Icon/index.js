import React from 'react'
import { default as PT } from 'prop-types'
import cx from 'classnames'

// Icons
import Add from './Add'
import Analytics from './Analytics'
import ArrowDown from './ArrowDown'
import ArrowDownFill from './ArrowDownFill'
import ArrowRight from './ArrowRight'
import ArrowUp from './ArrowUp'
import ArrowUpFill from './ArrowUpFill'
import ChevronDown from './ChevronDown'
import ChevronLeft from './ChevronLeft'
import ChevronRight from './ChevronRight'
import ChevronUp from './ChevronUp'
import Clock from './Clock'
import Close from './Close'
import Copy from './Copy'
import Error from './Error'
import Edit from './Edit'
import Help from './Help'
import Home from './Home'
import InformationCircle from './InformationCircle'
import Line from './Line'
import Link from './Link'
import NewWindow from './NewWindow'
import Null from './Null'
import Page from './Page'
import Plus from './Plus'
import Post from './Post'
import Search from './Search'
import Site from './Site'
import StaticPage from './StaticPage'
import Success from './Success'
import User from './User'
import Warning from './Warning'

export const ICONS = {
  add: Add,
  analytics: Analytics,
  arrowDown: ArrowDown,
  arrowDownFill: ArrowDownFill,
  arrowRight: ArrowRight,
  arrowUp: ArrowUp,
  arrowUpFill: ArrowUpFill,
  chevronDown: ChevronDown,
  chevronLeft: ChevronLeft,
  chevronRight: ChevronRight,
  chevronUp: ChevronUp,
  clock: Clock,
  close: Close,
  copy: Copy,
  edit: Edit,
  error: Error,
  help: Help,
  home: Home,
  informationCircle: InformationCircle,
  line: Line,
  link: Link,
  newWindow: NewWindow,
  null: Null,
  page: Page,
  plus: Plus,
  post: Post,
  search: Search,
  site: Site,
  staticPage: StaticPage,
  success: Success,
  user: User,
  warning: Warning,
}

const Icon = ({ name, size, status, className, ...props }) => {
  const SpecificIcon = ICONS[name]

  const styles = cx(
    'ola_icon',
    `is-${size}`,
    { 'ola-skeleton': status !== 'loaded' },
    { 'is-loading': status === 'loading' },
    className
  )

  if (status !== 'loaded') {
    return (
      <div className={styles}>
        <span></span>
      </div>
    )
  }

  return (
    <svg
      width="28"
      height="28"
      viewBox="0 0 28 28"
      fillRule="evenodd"
      className={styles}
      {...props}
    >
      <SpecificIcon />
    </svg>
  )
}

Icon.defaultProps = {
  name: 'help',
  size: 'medium',
  className: null,
  status: 'loaded',
}

Icon.propTypes = {
  /** Icons */
  name: PT.oneOf(Object.keys(ICONS)),
  /** Size */
  size: PT.oneOf(['small', 'medium', 'big']),
  /** Kpi status */
  status: PT.oneOf(['loaded', 'loading', 'empty']),
  /** Extra className */
  className: PT.string,
}

export default Icon
