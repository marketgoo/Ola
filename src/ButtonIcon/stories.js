import React from 'react'
import ButtonIcon from './'
import { Copy, ShieldCheck, Airplane } from '@phosphor-icons/react'

export default {
  title: 'ButtonIcon',
  component: ButtonIcon,
  args: {
    icon: 'home',
    color: 'neutral',
    as: 'button',
    variant: 'regular',
    invert: false,
    disabled: false,
    busy: false,
  },
  argTypes: {
    onClick: { action: 'clicked' },
    onFocus: { action: 'focused' },
    color: {
      control: {
        type: 'select',
        options: ['primary', 'neutral', 'positive', 'negative', 'warning', 'pro', 'accent'],
      },
    },
    icon: {
      control: {
        type: 'select',
        options: ['add', 'analytics', 'arrowDown', 'arrowRight', 'arrowUp', 'chevronDown', 'chevronLeft', 'chevronRight', 'chevronUp', 'clock', 'close', 'copy', 'error', 'edit', 'help', 'home', 'informationCircle', 'link', 'newWindow', 'null', 'page', 'plus', 'post', 'search', 'site', 'staticPage', 'success', 'user', 'warning'], 
      },
    },
  },
  decorators: [
    (Story) => (
      <div style={{ backgroundColor: '#f6f6ff', padding: '32px' }}>
        {Story()}
      </div>
    ),
  ],
}

export const Base = (args) => <ButtonIcon {...args} />

export const PositiveColor = (args) => <ButtonIcon {...args} />
PositiveColor.args = {
  color: 'positive'
}

export const ProColor = (args) => <ButtonIcon {...args} />
ProColor.args = {
  color: 'pro'
}

export const Disabled = (args) => <ButtonIcon {...args} />
Disabled.args = {
  disabled: true
}

export const Busy = (args) => <ButtonIcon {...args} />
Busy.args = {
  busy: true
}

export const Invert = (args) => <ButtonIcon {...args} />
Invert.args = {
  invert: true
}

export const VariantDark = (args) => <ButtonIcon {...args} />
VariantDark.args = {
  variant: 'dark'
}

export const WithPhosphorIcons = (args) => <>
  <ButtonIcon {...args} icon={<Copy />} color="primary" />
  <ButtonIcon {...args} icon={<ShieldCheck />} color="positive" />
  <ButtonIcon {...args} icon={<Airplane />} color="pro" />
</>

