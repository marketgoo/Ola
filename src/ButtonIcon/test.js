import React from 'react'
import ButtonIcon from './'
import renderer from 'react-test-renderer'

it('Default ButtonIcon', () => {
  const tree = renderer
    .create(
      <ButtonIcon icon="success" color="positive" />
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
