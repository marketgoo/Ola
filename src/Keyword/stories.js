import React from 'react'
import ButtonIcon from '../ButtonIcon'
import Icon from '../Icon'
import Keyword from './'

export default {
  title: 'Keyword',
  component: Keyword,
  args: {
    children: 'This is a keywordThis is a keywordThis is a keyword'
  }
}

export const Base = (args) => <Keyword {...args}>{args.children}</Keyword>

export const WithButtonIcon = (args) => <Keyword {...args} button={<ButtonIcon icon="close" />}>{args.children}</Keyword>
