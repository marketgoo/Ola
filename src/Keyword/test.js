import React from 'react'
import Keyword from './'
import ButtonIcon from '../ButtonIcon'
import Icon from '../Icon'
import renderer from 'react-test-renderer'

it('Default Keyword', () => {
  const tree = renderer.create(<Keyword>Test</Keyword>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Keyword with extraClass', () => {
  const tree = renderer.create(<Keyword className="test">Test</Keyword>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Keyword with button', () => {
  const tree = renderer.create(<Keyword className="test" button={<ButtonIcon icon="close" />}>Test</Keyword>).toJSON()
  expect(tree).toMatchSnapshot()
})
