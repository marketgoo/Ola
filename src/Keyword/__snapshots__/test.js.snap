// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Keyword 1`] = `
<strong
  className="ola_keyword"
>
  <span
    className="ola_keyword-text"
  >
    Test
  </span>
</strong>
`;

exports[`Keyword with button 1`] = `
<strong
  className="ola_keyword test"
>
  <span
    className="ola_keyword-text"
  >
    Test
  </span>
  <span
    className="ola_keyword-button"
  >
    <button
      className="ola_button_icon is-normal"
      disabled={false}
      style={
        Object {
          "--base-color": "var(--color-neutral-500)",
          "--busy-color": "var(--color-neutral-600)",
          "--disabled-color": "var(--color-neutral-300)",
          "--shadow-focus": "var(--shadow-focus-neutral)",
        }
      }
    >
      <svg
        className="ola_icon is-medium"
        fillRule="evenodd"
        height="28"
        viewBox="0 0 28 28"
        width="28"
      >
        <path
          d="M22.92 4c.296 0 .55.105.762.314.212.21.318.462.318.758s-.106.55-.318.762l-8.17 8.174 8.17 8.166c.212.212.318.464.318.754 0 .296-.106.549-.318.758-.212.21-.466.314-.762.314s-.547-.103-.753-.31l-8.171-8.166-8.163 8.166c-.206.207-.46.31-.761.31a1.04 1.04 0 0 1-.758-.31A1.02 1.02 0 0 1 4 22.936c0-.301.103-.555.31-.762l8.17-8.166-8.17-8.174A1.026 1.026 0 0 1 4 5.08c0-.296.105-.55.314-.762.21-.212.462-.318.758-.318.295 0 .55.106.761.318l8.163 8.174 8.17-8.174A1.03 1.03 0 0 1 22.92 4z"
        />
      </svg>
    </button>
  </span>
</strong>
`;

exports[`Keyword with extraClass 1`] = `
<strong
  className="ola_keyword test"
>
  <span
    className="ola_keyword-text"
  >
    Test
  </span>
</strong>
`;
